onyo/lib/filters.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+import re
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from onyo.lib import Repo
+
+
+# TODO: Move this to a place specifically meant for defaults, along with
+#  other defaults like <list>, <dict>, and potentially <none> or <null>
+UNSET_VALUE = '<unset>'
+
+
+class OnyoInvalidFilterError(Exception):
+    """Raise if filters are invalidly defined"""
+
+
+def asset_name_to_keys(path: Path, pseudo_keys: list[str]) -> dict[str, str]:
+    """Convert an asset name to pseudo key values"""
+    return dict(zip(pseudo_keys, re.split('[_.]', path.name)))
```

Comment:
I can change this, but I've again changed it in a PR that is currently blocked by this one. That PR creates a routine to handle retrieving pseudo keys and unifies all of the different approaches. Do you want me to change this right now, or are you willing to wait on the PR addressing this more thoroughly?

Reply:
I think for now it is sufficient to copy e.g.
`[serial, model, make, type] = [field[::-1] for field in re.findall(r'(.*)\.(.*)_(.*)_(.*)', asset.name[::-1])[0]]`
from https://github.com/psyinfra/onyo/blob/main/onyo/lib/onyo.py#L1324 to prevent the bug, and having the real solution added in the next PR is fine then :)

Reply:
The `Repo.valid_name()` function uses `re.findall('(^[^._]+?)_([^._]+?)_([^._]+?)\.(.+)', asset.name)[0]`. Is there a reason the one you're citing is different?

Reply:
Nope, just the classic problem that the name scheme/pseudo key functionality is neither unified nor fully defined, good catch!

Reply:
I'll use the one I cited then, and this one should then also be used when making a unified `get_pseudo_keys` function.